tests/validation/tests/v3_api/common.py

```diff
@@ -2311,3 +2315,52 @@ def auth_resource_cleanup():
         user_crtbs = client.list_cluster_role_template_binding(userId=user.id)
         for crtb in user_crtbs:
             client.delete(crtb)
+
+
+class WebsocketLogParse:
+    """
+    the class is used for receiving and parsing the message
+    received from the websocket
+    """
+    def __init__(self):
+        self.lock = Lock()
+        self._last_message = ''
+
+    def receiver(self, socket, skip):
+        """
+        run a thread to receive and save the message from the web socket
+        :param socket: the socket connection
+        :param skip: if True skip the first char of the received message
+        """
+        while True:
+            try:
+                data = socket.recv()
+                # the message from the kubectl contains an extra char
+                if skip:
+                    data = data[1:]
+                if len(data) < 5:
+                    pass
+                data = base64.b64decode(data).decode()
+                self.lock.acquire()
+                self._last_message += data
+                self.lock.release()
+            except websocket.WebSocketConnectionClosedException:
+                print("Connection closed")
+                break
+
+    @staticmethod
```

Comment:
We would probably remove the static of this method and make it non-static if we end up making a `finish_thread` method.